### PR TITLE
[RGPD] Add new checkbox in form and data on user registration

### PR DESCRIPTION
**Pull request by JeremieSamson**
Capture du 2019-06-01 23-07-17

### CODE REVIEW

File: src/Migrations/Version20190601203337.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20190601203337 extends AbstractMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE user ADD privacy_accepted TINYINT(1) NOT NULL, ADD privacy_accepted_at DATETIME DEFAULT NULL, ADD privacy_accepted_with LONGTEXT DEFAULT NULL');
+        $this->addSql('UPDATE user SET privacy_accepted = false');
```
Comment (JeremieSamson) [inline src/Migrations/Version20190601203337.php]: All previous user didn't accept the privacy policy explicitly

File: translations/messages.en.yml [hunk 2]
```diff
@@ -43,6 +43,7 @@ registration.city: 'Your (exchange) city for the next semester'
 registration.city.placeholder: 'Select a City...'
 registration.signin: 'Sign up'
 registration.title: 'Sign up'
+registration.data_use: 'I agree that the data filled in this form will be used by the BuddySystem to find a person who best corresponds to me in accordance with the <a href="%url%">Privacy Policy</a>'
```
Comment (JeremieSamson) [inline translations/messages.en.yml]: Maybe we can change this

File: src/Entity/User.php [hunk 3]
```diff
@@ -133,6 +133,27 @@ class User extends BaseUser implements ParticipantInterface
...
+     * @var \DateTime|null
+     *
+     * @ORM\Column(type="datetime", nullable=false, nullable=true)
+     */
+    private $privacyAcceptedAt;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="text", nullable=false, nullable=true)
```
Comment (JeremieSamson) [inline src/Entity/User.php]: nullable because we didn't catch user registration info for previous user

File: translations/messages.en.yml [hunk 4]
```diff
@@ -43,6 +43,7 @@ registration.city: 'Your (exchange) city for the next semester'
 registration.city.placeholder: 'Select a City...'
 registration.signin: 'Sign up'
 registration.title: 'Sign up'
+registration.data_use: 'I agree that the data filled in this form will be used by the BuddySystem to find a person with a profile as similar as possible, in accordance with the <a href="%url%">Privacy Policy</a>'
```
Comment (victor-paumier) [inline translations/messages.en.yml]: put tags in twig, not in translation

Comment (JeremieSamson) [reply translations/messages.en.yml]: done

**Pull request merged**